docs/proposals/config_data.md

```diff
@@ -0,0 +1,327 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
```

Comment:
We want to use it to configure access to services. For example, you could create a ConfigData resource that contains information about connecting to an Oracle database, such as its SID, and that isn't something you can set in a Service.